### ISSUE TITLE
[Q] Should add a max memory limit on `bf.reserve`

### ISSUE THREAD

**Issue by arthurkiller** [RedisBloom/RedisBloom#699]
`bf.reserve test 0.01 1000000000000000000`

bf.reserve can cause server OOM, should we add a limitation on max memory size?

https://github.com/RedisBloom/RedisBloom/blob/master/src/rebloom.c#L142C5

**Comment by LiorKogan** [comment]
@arthurkiller thank you,
Indeed, we need to fail gracefully if the allocation size is not realistic (a single Redis key's value should be limited to 512MB).

**Comment by arthurkiller** [comment]
> @arthurkiller thank you, Indeed, we need to fail gracefully if the allocation size is unrealistic (a single Redis key's value should be limited to `proto-max-bulk-len` (512MB)).

@LiorKogan  thx for your reply, I can help with this pr

**Comment by arthurkiller** [comment]
> @arthurkiller thank you, Indeed, we need to fail gracefully if the allocation size is unrealistic (a single Redis key's value should be limited to `proto-max-bulk-len` (512MB)).

There are two ways to fix this issue
1. try to limit the key size below `proto-max-bulk-len` by reading redis server config
2. set another threshold inside bf module, maybe a macro？

What do you think about that @LiorKogan

**Comment by ashtul** [comment]
@arthurkiller The key is not limited intentionally. The limit of 512MB is only applied when you try to retrieve the key.
With probabilistic data-structures you will need this only when you download the filter for backup or to transfer it to another computer.
This is the reason BF has `BF.SCANDUMP` and `BF.LOADCHUNK`.

**Comment by arthurkiller** [comment]
> @arthurkiller The key is not limited intentionally. The limit of 512MB is only applied when you try to retrieve the key. With probabilistic data-structures you will need this only when you download the filter for backup or to transfer it to another computer. This is the reason BF has `BF.SCANDUMP` and `BF.LOADCHUNK`.

@ashtul exactly. but please consider providing RB to users as a part of Redis products. once a user tries to call `bf.reserve` with a huge buffer size by mistake then the server crashes down immediately without any warranty or feedback.

Shouldn't we make sure the command can give back an error or just get some protection to make it more safe to use?

**Comment by arthurkiller** [comment]
server should not crash in any case by calling just a command, only if this could do harm to the core data or we need to panic in this situation.

**Comment by arthurkiller** [comment]
Any new suggestions on this?
